pkg/csi/service/vanilla/controller.go

```diff
@@ -1330,9 +1352,11 @@ func (c *controller) createBlockVolumeWithPlacementEngineForMultiVC(ctx context.
 		}
 	}
 	if volumeInfo == nil {
+		if faultType == "" {
```

Comment:
Why do we need to check for fault and then set internalFault.
Are the internal functions not returning the expected faults ?

Reply:
Yes in some cases while provisioning volume, if we can't provision volume on the first vCenter, we go and attempt to provision volume on the second vCenter, when we continue to the next VC, we do not set any fault. and in the end if none of the VC can provision volume, we might get nil volumeMgr with no fault. 

example is when SPBM policy is not found in any of the vCenters, we collect logs in `combinedErrMssgs` and in the end if volumeInfo is nil, we return error with all combinedErrMssgs by setting fault type as `CSIInternalFault`.

```
    Warning ProvisioningFailed 4s (x3 over 7s) csi.vsphere.vmware.com_vsphere-csi-controller-75fc75c546-s99n4_1c0fd856-b4e3-4fab-a86c-1390931e4a16 failed to provision volume with StorageClass "block-sc": rpc error: code = Internal desc = failed to create volume. Errors encountered: [Storage policy name "wrong-policy-name" not found in VC "sc1-10-78-81-17.eng.vmware.com" Storage policy name "wrong-policy-name" not found in VC "sc2-10-185-105-221.nimbus.eng.vmware.com"]

```